### ISSUE TITLE
getCallRecordings returns callAsText instead of callAsStream by default

### ISSUE THREAD

**Issue by dsolowitz** [Bandwidth/node-voice#49]
### Description
I'm not sure it's a bug, but more of a usability issue.... From the customer:
We're trying to use the NodeJS SDK version 4.1.2 (from NPM "@bandwidth/voice": "^4.1.2") to download a recording. The credentials and recording information passed to the SDK are correct, as we've successfully downloaded it using cURL.

When we call the getDownloadCallRecording method, the data returned is a binary string. We tried to encode the string into a wav file, but something goes wrong in the process, as VLC can't play the resulting file, whereas it can play it normally when we download it with cURL.

We noticed that if we change the getDownloadCallRecording method to return a buffer instead of a string, return req.callAsStream(requestOptions) instead of return req.callAsText(requestOptions), we can also save the audio data to a file in a way VLC can play it.

Basically I think this should default to a stream for ease of use.

### Environment Information

 - OS Version: ???
 - SDK Version:  "@bandwidth/voice": "^4.1.2"
 - Environment: Node

### Sample Code Snippet
Snippet provided above
```

### Stack Trace
<!-- Please provide the stack trace that shows the bug. -->
```shell
# Stack Trace
```

### Expected Behavior
Default to a stream instead of text.

### Suggested Fix
I'm not sure the fix, but the requistBuilder template for the SDK has the option for streaming, so I'm not sure how to set the default for that particular method seen here: https://github.com/Bandwidth/node-voice/blob/9f113bf2570d8b8ff03dc162dd36de5da74939fc/src/http/requestBuilder.ts#L79

**Comment by bernardodesousa** [comment]
I believe you're referring to the getDownloadCallRecording method, not getCallRecordings. I think getCallRecordings returns json, as it should.

**Comment by dsolowitz** [comment]
He's correct, in my haste I wrote the wrong method name down, this is the correct one:

async getDownloadCallRecording(
    accountId: string,
    callId: string,
    recordingId: string,
    requestOptions?: RequestOptions
  ): Promise<ApiResponse<unknown>> {
    const req = this.createRequest('GET');
    req.baseUrl('VoiceDefault');
    const mapped = req.prepareArgs({
      accountId: [accountId, string()],
      callId: [callId, string()],
      recordingId: [recordingId, string()],
    });
    req.appendTemplatePath`/api/v2/accounts/${mapped.accountId}/calls/${mapped.callId}/recordings/${mapped.recordingId}/media`;
    req.throwOn(400, ExternalApiError, 'Something\'s not quite right... Your request is invalid. Please fix it before trying again.');
    req.throwOn(401, ApiError, 'Your credentials are invalid. Please use your Bandwidth dashboard credentials to authenticate to the API.');
    req.throwOn(403, ExternalApiError, 'User unauthorized to perform this action.');
    req.throwOn(404, ExternalApiError, 'The resource specified cannot be found or does not belong to you.');
    req.throwOn(415, ExternalApiError, 'We don\'t support that media type. If a request body is required, please send it to us as `application/json`.');
    req.throwOn(429, ExternalApiError, 'You\'re sending requests to this endpoint too frequently. Please slow your request rate down and try again.');
    req.throwOn(500, ExternalApiError, 'Something unexpected happened. Please try again.');
    **return req.callAsText(requestOptions);**
  }

Highlighted where callAsText is returned instead of Stream

**Comment by ajrice6713** [comment]
Hey All,

Thanks for raising this up. let me do some testing and get back to you on this - with how we return the binary from the API im not sure off the top of my head if [callAsStream](https://github.com/Bandwidth/node-voice/blob/9f113bf2570d8b8ff03dc162dd36de5da74939fc/src/http/requestBuilder.ts#L157) will work but we will confirm and push a fix if so! 

Unfortunately since recordings delete its hard for us to write programatic tests between 2 computers to have recordings that we can fetch/delete - but a more robust testing strategy for that is on our roadmap.

**Comment by ajrice6713** [comment]
Hey @bernardodesousa,

I have a PR open to resolve this - #51 

@ckoegel and I have tested recording calls and conferences then getting the media using the SDK on branch DX-2531 and postman and comparing the hashmaps of the 2 files, and everything looks good 

We will release a beta tomorrow (4/12/22) from that branch and then hopefully begin scoping the work for the dynamic testing pipeline next sprint. 

Let us know if you have any issues with the branch of the beta tomorrow.

Code sample using the callAsStream response: 

```node
  const callId = req.body.callId;
  const recordingId = req.body.recordingId;
  try {
      const recordingStream = await controller.getDownloadCallRecording(accountId, callId, recordingId);
      var writeStream = fs.createWriteStream('recording.wav', { flags : 'w' });
      recordingStream.body.pipe(writeStream);
  } catch(e) {
      console.log(e)
      res.status(500).send(e.message);
  }
```

**Comment by bernardodesousa** [comment]
Hi! How should we use the beta version? I can't seem to find it.

**Comment by ajrice6713** [comment]
Hey @bernardodesousa - we hit a snag with deploying betas - I sent an email yesterday but you might not have been a recipient 

Should have an offical npm beta release in a few days while we build a beta-release specific workflow - for the time being are you able to test the package on the DX-2531 branch? Or install from git directly using 
```
npm install "https://github.com/Bandwidth/node-voice.git#DX-2531" --save
```

**Comment by bernardodesousa** [comment]
Alright, thanks @ajrice6713

**Comment by bernardodesousa** [comment]
Hello! The package isn't built on a Bitbucket pipeline, while it is built correctly locally.

We list `"@bandwidth/voice": "git+https://git@github.com/Bandwidth/node-voice.git#DX-2531"` in our dependencies, in package.json. When we run `npm i` locally, the package is correctly built. The file in node_modules/@bandwidth/voice/dist/index.js exists. In the pipeline, however, we get this error:

`Error: Cannot find module '/opt/atlassian/pipelines/agent/build/node_modules/@bandwidth/voice/dist/index.js'`

We tried a bunch a options:

git+https://github.com/Bandwidth/node-voice.git#DX-2531
git+https://git@github.com/Bandwidth/node-voice.git#DX-2531
git+https://git@github.com/Bandwidth/node-voice.git#dbaaa542c3944ecb5c44fa3495c13a38a5ddc587
git+https://git@github.com/Bandwidth/node-voice.git

All result in the same error.

Interestingly, we already had a dependency defined by a git URL (git+https://git@github.com/bcosma/winston-papertrail.git) and that package is built both locally on our machines and on the pipeline.

Any idea on what might be wrong?

**Comment by ajrice6713** [comment]
Hey @bernardodesousa,

Im not sure as to why that would happen - I am not familiar with BitBucket

We did however just publish `v4.1.2-beta.0` to NPM, it should be live and available to you for testing! 

```sh
npm i @bandwidth/voice@4.2.1-beta.0
```

**Comment by bernardodesousa** [comment]
Hi! PR 51 was merged into v4.2.1-beta.0, but those changes didn't seem to have ever made it into the main line. The method getDownloadCallRecording used to return json, then I raised an issue that @ckoegel fixed in PR 44, and that's when the method started returning a binary string. However, we kept using v4.2.1-beta.0, in which a stream is returned. That branch became stale.

So my question is: should the return value of getDownloadCallRecording be treated as a stream in 5.4.0?
